File: src/mysqlcookbook/recipes/metadata/get_column_info.rb
#!/usr/bin/ruby
# get_column_info.rb: wrapper to demonstrate get_column_info()
# utility routine

# Assumes that you've created the "image" table!

require "Cookbook"
require "Cookbook_Utils"

db_name = "cookbook"
tbl_name = "image"

begin
  client = Cookbook.connect
rescue Mysql2::Error => e
  puts "Could not connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
end

puts "Using get_column_info()"
puts "Column information for #{db_name}.#{tbl_name} table:"
info = get_column_info(client, db_name, tbl_name)
info.each do |col_name, col_info|
  puts "  Column: #{col_name}"
  col_info.each do |col_val_key, col_val|
    puts "    #{col_val_key}: #{col_val}"
  end
end

client.close

File: src/mysqlcookbook/recipes/metadata/get_column_info.py
#!/usr/bin/python3
# get_column_info.py: wrapper to demonstrate get_column_info()
# utility routine

# Assumes that you've created the "image" table!

import sys
import mysql.connector
import cookbook
from cookbook_utils import *

db_name = "cookbook"
tbl_name = "image"

try:
  conn = cookbook.connect()
except mysql.connector.Error as e:
  print("Error code: %s" % e.errno)
  print("Error message: %s" % e.msg)
  sys.exit(1)

print("Using get_column_info()")
print("Column information for %s.%s table:" % (db_name, tbl_name))
info = get_column_info(conn, db_name, tbl_name)
for col_name in info:
  col_info = info[col_name]
  print("  Column: %s" % col_name)
  for col_info_key in col_info:
    print("    %s: %s" % (col_info_key, col_info[col_info_key]))

conn.close()
